### ISSUE TITLE
TESTING: Run selection of travis tests using valgrind

### ISSUE THREAD

**Issue by jhamman** [UW-Hydro/VIC#492]
As discussed in https://github.com/UW-Hydro/VIC/pull/490#issuecomment-216582270, it would be good to run a selection of our travis tests with `valgrind`.  This will catch memory errors and leaks like what was reported in #463. We basically need to run a command that looks something like this and then check for the `VIC_VALGRIND_ERR` error code after the test.

```Bash
VIC_VALGRIND_ERR=-1
valgrind -v --leak-check=full --error-exitcode=$VIC_VALGRIND_ERR vic_image.exe -g global_file
```
 

xref: #442 #463 https://github.com/UW-Hydro/VIC/pull/490#issuecomment-216582270

**Comment by jhamman** [comment]
xref: https://github.com/UW-Hydro/tonic/pull/53

**Comment by jhamman** [comment]
closed via #506.
